app/src/org/commcare/activities/DrfitHelper.java

```diff
@@ -0,0 +1,73 @@
+package org.commcare.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.support.v7.preference.PreferenceManager;
+
+import org.commcare.CommCareApplication;
+import org.commcare.core.services.CommCarePreferenceManagerFactory;
+import org.commcare.dalvik.R;
+import org.commcare.preferences.DeveloperPreferences;
+import org.commcare.preferences.PrefValues;
+import org.commcare.views.dialogs.StandardAlertDialog;
+
+import java.util.Date;
+
+import static org.commcare.core.network.CommCareNetworkServiceGenerator.CURRENT_DRIFT;
+import static org.commcare.core.network.CommCareNetworkServiceGenerator.MAX_DRIFT_SINCE_LAST_HEARTBEAT;
+import static org.javarosa.core.model.utils.DateUtils.DAY_IN_MS;
+
+public class DrfitHelper {
+
+    private static final String INCORRECT_TIME_WARNING_ENABLED = "incorrect_time_warning_enabled";
+    private static final String LAST_INCORRECT_TIME_WARNING_AT = "last_incorrect_time_warning_at";
+
+    static boolean shouldShowDriftWarning() {
+        if (isWarningEnabled()) {
+            long lastIncorrrectTimeWarningAt = getAppPreferences().getLong(LAST_INCORRECT_TIME_WARNING_AT, -1);
+            if (new Date().getTime() - lastIncorrrectTimeWarningAt > DAY_IN_MS) {
```

Comment:
Should this threshold be configurable at all? How did we decide on a day?

Reply:
I just took it from Clayton's initital spec of the feature and Derek didn't seem to have any issues with this. It just looks like a good duration to start with.

Reply:
Fair